### DISCUSSION TITLE
Mastodon does not log at all

### DISCUSSION THREAD

**Discussion by M-Stenzel** [mastodon/mastodon#25573] [General]
Hi, 
this is a Linux (non docker!) installation of mastodon 4.1.2.

Mastodon does not log, I have this log directory with 2 files (.keep & production.log), both files are empty.
/var/log/messages and /var/log/warn do not contain any information regarding mastodon.

These are the systemd files:

```
[Unit]
Description=mastodon-sidekiq
After=network.target

[Service]
Type=simple
User=mastodon
WorkingDirectory=/home/mastodon/live
Environment="RAILS_ENV=production"
Environment="DB_POOL=25"
Environment="MALLOC_ARENA_MAX=2"
Environment="RAILS_LOG_LEVEL=info"
ExecStart=/home/mastodon/.rbenv/shims/bundle exec sidekiq -c 25
TimeoutSec=15
Restart=always

[Install]
WantedBy=multi-user.target
```

and 
```

[Unit]
Description=mastodon-streaming
After=network.target

[Service]
Type=simple
User=mastodon
WorkingDirectory=/home/mastodon/live
Environment="NODE_ENV=production"
Environment="PORT=6000"
Environment="STREAMING_CLUSTER_NUM=1"
Environment="RAILS_LOG_LEVEL=info"
ExecStart=/usr/bin/node16 ./streaming
TimeoutSec=15
Restart=always

[Install]
WantedBy=multi-user.target
```

and
```

[Unit]
Description=mastodon-web
After=network.target

[Service]
Type=simple
User=mastodon
WorkingDirectory=/home/mastodon/live
Environment="RAILS_ENV=production"
Environment="RAILS_LOG_LEVEL=info"
Environment="PORT=3030"
ExecStart=/home/mastodon/.rbenv/shims/bundle exec puma -C config/puma.rb
ExecReload=/bin/kill -SIGUSR1 $MAINPID
TimeoutSec=15
Restart=always

[Install]
WantedBy=multi-user.target
```

HOW can I enable logging? The variable "RAILS_LOG_LEVEL" does not seem to change anything.

Martin.

P. S. Write permissions to the directory and files for user "mastodon" are fulfilled.

**Comment by Gargron** [comment]
Check `journalctl -u mastodon-web` etc

**Comment by M-Stenzel** [comment]
This is the result


-- Logs begin at Thu 2022-12-22 22:10:13 EET, end at Sun 2023-06-25 20:03:15 EEST. --
-- No entries --

P. S. I set up the instance on Dec. 22 2022.

**Reply by Gargron** [reply]
Try `systemctl restart systemd-journald`

**Reply by M-Stenzel** [reply]
No change... thank you!

**Reply by Gargron** [reply]
Mastodon logs to stdout. Systemd is responsible for capturing stdout and writing it to its journaling system. I would investigate your systemd configuration (not the service definitions for Mastodon, but logging configuration in journald etc).
